web/src/settings_users.js

```diff
@@ -440,7 +440,10 @@ export function update_user_data(user_id, new_data) {
     }
 
     if (new_data.role !== undefined) {
-        $user_row.find(".user_role").text(people.get_user_type(user_id));
+        const user_type = people.get_user_type(user_id);
+        if (user_type) {
+            $user_row.find(".user_role").text(user_type);
+        }
```

Comment:
Given this implementation, we probably can cut out the possibility of it being `undefined` by using `new_data.role` with `settings_config.user_role_map` and asserting the role is valid. But that can be a follow-up.

```
export function get_user_type(user_id: number): string | undefined {                                    
    const user_profile = get_by_user_id(user_id);                                                       
    return settings_config.user_role_map.get(user_profile.role);                                        
}                                                                                                       
```

Reply:
There are quite a few places that user `people.get_user_type` and I'm pretty wary to change its return type without doing a deeper dive on how all those places handle the undefined value. I feel like this could be a better followup when we're further down the typescript conversion and can easily find all those instances. I'll make a note to return to this later.

Reply:
To be clear, the thought was more that this particular caller could just access `settings_config.user_role_map` directly rather than using `get_user_type`.